File: src/main/java/no/nav/fo/veilarbregistrering/orgenhet/Norg2Gateway.java
package no.nav.fo.veilarbregistrering.orgenhet;

import no.nav.fo.veilarbregistrering.enhet.Kommunenummer;

import java.util.Map;
import java.util.Optional;

public interface Norg2Gateway {

    Optional<Enhetnr> hentEnhetFor(Kommunenummer kommunenummer);

    Map<Enhetnr, NavEnhet> hentAlleEnheter();
}

File: src/main/java/no/nav/fo/veilarbregistrering/orgenhet/Norg2Gateway.kt
package no.nav.fo.veilarbregistrering.orgenhet

import no.nav.fo.veilarbregistrering.enhet.Kommune
import java.util.*

interface Norg2Gateway {
    fun hentEnhetFor(kommune: Kommune): Optional<Enhetnr>
    fun hentAlleEnheter(): Map<Enhetnr, NavEnhet>
}